src/binder/expression/agg_call.rs

```diff
@@ -89,10 +89,16 @@ impl Binder {
             }
         }
         let (kind, return_type) = match func.name.to_string().to_lowercase().as_str() {
-            "avg" => (
-                AggKind::Avg,
-                Some(DataType::new(DataTypeKind::Double, false)),
-            ),
+            "avg" => {
+                let agg_kind = AggKind::Avg;
+                let mut default_data_type = Some(DataType::new(DataTypeKind::Double, false));
+                if let Some(ref data_type) = args[0].return_type() {
+                    if let DataTypeKind::Decimal(_, _) = data_type.kind {
+                        default_data_type = args[0].return_type();
+                    }
+                }
+                (agg_kind, default_data_type)
+            }
```

Comment:
Could you explain a bit why to change this (in this PR)? It seems not relevant. 🤔

Reply:
For example in q1.sql.
```sql
select
    l_returnflag,
    l_linestatus,
    sum(l_quantity) as sum_qty,
    sum(l_extendedprice) as sum_base_price,
    sum(l_extendedprice * (1 - l_discount)) as sum_disc_price,
    sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)) as sum_charge,
    avg(l_quantity) as avg_qty,
    avg(l_extendedprice) as avg_price,
    avg(l_discount) as avg_disc,
    count(*) as count_order
from
    lineitem
where
    l_shipdate <= date '1998-12-01' - interval '71' day
group by
    l_returnflag,
    l_linestatus
order by
    l_returnflag,
    l_linestatus;
```
For `avg(l_quantity) as avg_qty`、`avg(l_extendedprice) as avg_price`、`avg(l_discount) as avg_disc`，the evaluate result is `DataTypeKind::Decimal`, But The output_type of excuter will be `DataTypeKind::Double`. So it will panic when DataChunkBuilder.push_row(row.values) in `order.rs`.

Reply:
I think the problem arose because when we rewrite `avg` into `sum / count` (a few lines below, starting from the `match kind`), the `return_type` mismatches `left_expr`'s type. Would it be better to fix the problem there?